State.h

```diff
@@ -0,0 +1,44 @@
+#ifndef STATE_H
+#define STATE_H
+
+#include <Arduino.h>
+#include <math.h>
+
+class State {
+private:
+    float x, y, z;    // Cartesian coordinates
+    float theta, phi; // Spherical coordinates
```

Comment:
It might be more straightforward to just have one single representation as two complex numbers representing the ket and having this conversion back and forth in the getters and setters. Especially when we start working with entanglement, the ket will be the main object with which to work internally.

Reply:
This is not done yet but will work on it in coming days. Should there be 3 different constructors (eg. for cartesian, spherical and complex) or its better to have one and just keep the conversion inside the setters/getters?